backend/ttrs/serializers.py

```diff
@@ -3,48 +3,22 @@
 from django.contrib.auth.password_validation import validate_password
 from rest_framework import serializers
 from rest_framework.exceptions import ValidationError
-from django.core.exceptions import ValidationError as DjangoValidationError
+from django.core.exceptions import ValidationError as DjangoValidationError, ObjectDoesNotExist
 
-from .models import Student, College, Department, Major, Course, Lecture, Evaluation, TimeTable
+from .models import Student, College, Department, Major, Course, Lecture, Evaluation, TimeTable, MyTimeTable, \
+    BookmarkedTimeTable, ReceivedTimeTable, TimeSlot, Classroom
 
 
 class StudentSerializer(serializers.ModelSerializer):
-    my_time_table = serializers.SerializerMethodField()
-    bookmarked_time_tables = serializers.SerializerMethodField()
-    received_time_tables = serializers.SerializerMethodField()
-
     class Meta:
         model = Student
         fields = ('id', 'username', 'password', 'email', 'grade', 'college', 'department', 'major', 'not_recommends',
-                  'my_time_table', 'bookmarked_time_tables', 'received_time_tables')
+                  'my_time_tables', 'bookmarked_time_tables', 'received_time_tables')
+        read_only_fields = ('not_recommends', 'my_time_tables', 'bookmarked_time_tables', 'received_time_tables')
```

Comment:
From my_time_table**s**, I suppose you've decided to return multiple my tts. Right?

Reply:
Yes, but there will be only one my time table with certain year and semester or no table.